pkg/operator/sync_openshiftcontrollermanager_v311_00.go

```diff
@@ -282,9 +303,58 @@ func manageRouteControllerManagerClientCA_v311_00_to_latest(client coreclientv1.
 }
 
 // similar logic for route-controller-manager in manageRouteControllerManagerConfigMap_v311_00_to_latest
-func manageOpenShiftControllerManagerConfigMap_v311_00_to_latest(kubeClient kubernetes.Interface, client coreclientv1.ConfigMapsGetter, recorder events.Recorder, operatorConfig *operatorapiv1.OpenShiftControllerManager) (*corev1.ConfigMap, bool, error) {
+func manageOpenShiftControllerManagerConfigMap_v311_00_to_latest(clusterVersionInformer configinformersconfigv1.ClusterVersionInformer, kubeClient kubernetes.Interface, client coreclientv1.ConfigMapsGetter, recorder events.Recorder, operatorConfig *operatorapiv1.OpenShiftControllerManager) (*corev1.ConfigMap, bool, error) {
+	clusterVersion, err := clusterVersionInformer.Lister().Get("version")
+	if err != nil {
+		return nil, false, err
+	}
+	knownCaps := make(map[string]bool)
```

Comment:
https://github.com/openshift/cluster-openshift-apiserver-operator/pull/532/commits/2ca7323b1eb017da581135e02dadec626d22b4e3#diff-0d623dfd885adb20f991bda4c2453aebd732ca6dbb4d1d4be6e79805c3b48de6R409-R418 simplifies the detection of which caps are enabled.

Reply:
Tried moving it here. Please check :)